slither/tools/read_storage/__main__.py

```diff
@@ -126,22 +126,21 @@ def main() -> None:
     else:
         contracts = slither.contracts
 
-    srs = SlitherReadStorage(contracts, args.max_depth)
-
-    try:
-        srs.block = int(args.block)
-    except ValueError:
-        srs.block = str(args.block or "latest")
-
+    rpc_info = None
     if args.rpc_url:
-        # Remove target prefix e.g. rinkeby:0x0 -> 0x0.
-        address = target[target.find(":") + 1 :]
-        # Default to implementation address unless a storage address is given.
-        if not args.storage_address:
-            args.storage_address = address
-        srs.storage_address = args.storage_address
-
-        srs.rpc = args.rpc_url
+        try:
+            block = int(args.block)
+        except ValueError:
+            block = "latest"
```

Comment:
See also https://github.com/crytic/slither/pull/1843/files#r1185253313

Reply:
`block` is no longer a member of `SlitherReadStorage`, since it was moved to the new `RpcInfo` class. So how about this?
```python
    if args.rpc_url:
        if args.block in ["latest", "earliest", "pending", "safe", "finalized"]:
            block = args.block
        else:
            try:
                block = int(args.block)
            except ValueError:
                block = "latest"
        rpc_info = RpcInfo(args.rpc_url, block)
```

Reply:
```
    if args.rpc_url:
            try:
                block = int(args.block)
            except ValueError:
                block = BlockTag(args.block) or "latest"
        rpc_info = RpcInfo(args.rpc_url, block)
```

Reply:
Done!